File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/managers/entity/IEnemyEntityManager.java
package com.dkellycollins.triggerfinger.managers.entity;

import com.dkellycollins.triggerfinger.data.entity.IEnemy;
import com.dkellycollins.triggerfinger.data.model.IVector;

public interface IEnemyEntityManager {

    Iterable<IEnemy> retrieve();
    IEnemy retrieve(int enemyId);
    IEnemy retrieveByCollidable(int collidableId);

    int create(IVector position);

    void delete(int playerId);

}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/managers/entity/IEnemyEntityManager.kt
package com.dkellycollins.triggerfinger.managers.entity

import com.dkellycollins.triggerfinger.data.entity.IEnemy
import com.dkellycollins.triggerfinger.data.model.IVector

interface IEnemyEntityManager {

    fun retrieve(): Iterable<IEnemy>
    fun retrieve(enemyId: Int): IEnemy
    fun retrieveByCollidable(collidableId: Int): IEnemy

    fun create(position: IVector): Int

    fun delete(playerId: Int)

}
